test/c/test_target.c

```diff
@@ -0,0 +1,357 @@
+    if (result != QkExitCode_Success) {
+        printf("Unexpected error occurred when adding entry.");
+    }
+
+    // Test length
+    const size_t new_length = qk_target_entry_num_properties(property_map);
+    if (new_length != 1) {
+        printf("The initial length of the provided property map was not 1: %zu", length);
+        return EqualityError;
+    }
+
+    // Add invalid qargs
+    // Add some qargs and properties
+    uint32_t invalid_qargs[3] = {0, 1, 2};
+
+    QkExitCode error_result =
+        qk_target_entry_add_property(property_map, invalid_qargs, 3, 0.00018, 0.00002);
+    if (error_result != QkExitCode_TargetQargMismatch) {
+        printf("The operation did not fail as expected for invalid qargs.");
+    }
+
+    qk_target_entry_free(property_map);
+    return Ok;
+}
+
+/**
+ * Test adding an instruction to the Target.
+ */
+int test_target_add_instruction(void) {
+    const uint32_t num_qubits = 1;
+    // Let's create a target with one qubit for now
+    QkTarget *target = qk_target_new(num_qubits);
+    int result = Ok;
+    // Add an X Gate.
+
+    // This operation is global, no property map is provided
+    QkExitCode result_x = qk_target_add_instruction(target, qk_target_entry_new(QkGate_X));
+    if (result_x != QkExitCode_Success) {
+        printf("Unexpected error occurred when adding a global X gate.");
+        result = EqualityError;
+        goto cleanup;
+    }
+
+    // Re-add same gate, check if it fails
+    QkExitCode result_x_readded = qk_target_add_instruction(target, qk_target_entry_new(QkGate_X));
+    if (result_x_readded != QkExitCode_TargetInstAlreadyExists) {
+        printf("The addition of a repeated gate did not fail as expected.");
+        result = EqualityError;
+        goto cleanup;
+    }
+
+    // Number of qubits of the target should not change.
+    uint32_t current_num_qubits = qk_target_num_qubits(target);
+    if (current_num_qubits != 1) {
+        printf("The number of qubits this target is compatible with is not 1: %u",
+               current_num_qubits);
+        result = EqualityError;
+        goto cleanup;
+    }
+    size_t current_size = qk_target_num_instructions(target);
+    if (current_num_qubits != 1) {
+        printf("The size of this target is not correct: Expected 1, got %zu", current_size);
+        result = EqualityError;
+        goto cleanup;
+    }
+
+    // Add a CX Gate.
+    // Create prop_map for the instruction
+    // Add property for (0, 1)
+    QkTargetEntry *cx_entry = qk_target_entry_new(QkGate_CX);
+    uint32_t qargs[2] = {0, 1};
+    double inst_error = 0.0090393;
+    double inst_duration = 0.020039;
+
+    QkExitCode result_cx_props =
+        qk_target_entry_add_property(cx_entry, qargs, 2, inst_duration, inst_error);
+    if (result_cx_props != QkExitCode_Success) {
+        printf("Unexpected error occurred when adding property to a CX gate entry.");
+        return EqualityError;
+        goto cleanup;
```

Comment:
This doesn't look quite right (below as well), did you mean to set the result to `EqualityError`? Also we need to free `cx_entry` if we terminate early